CDCgov/trusted-intermediary
# DAST Scanning

# Dynamic Application Security Test 


This PR adds DAST scanning to our application. The scanning is done as part of a workflow. The event gets triggered when pushing to the main branch. 
Currently, the workflow fires up when pushing to the remote branch, this was done on purpose in order to help with testing. Once the PR gets approved, the DAST scanning will get triggered every time something gets pushed to the main branch.

## Changes
New Files
- health-check.sh
- dast.yml

health-check.sh - Checks to see if the api is functional by accessing the health endpoint.

dast.yml - Workflow that uses OWASP ZAP for DAST scanning

## Testing
You want to test the script and also the workflow.

### Testing the script:
- run shadowJar: `./gradlew shadowJar`
- build docker container: `docker compose up --build -d`
- run the script: `./health-check.sh`

### Testing the workflow:
The easiest way to do this would be to add a minor change to the branch and then push it.
Check the Actions on github and make sure everything passes.
There should be an artifact that contains warnings such as 404 status codes. 

## Issue
#77.
